WeatherApp/Services/NetworkManager.swift

```diff
@@ -1,19 +1,24 @@
 import Foundation
 
 class NetworkManager {
-    
-    func fetchFromApi(completionHandler:  @escaping (WeatherInfoData) -> Void) {
+    func fetchWeatherData(latitude: Double, longitude: Double, completionHandler:  @escaping (WeatherInfoData?) -> Void) {
         DispatchQueue.global(qos: .userInitiated).async {
-            let url = URL(string: "https://api.openweathermap.org/data/2.5/weather?lat=23.810331&lon=90.412521&units=metric&appid=f7fba2a96004431c6e3b90fb0728bd89")
-            let dataTask = URLSession.shared.dataTask(with: url!) { (data, response, error) in
+            let url = URL(string: "https://api.openweathermap.org/data/2.5/weather?lat=\(latitude)&lon=\(longitude)&units=metric&appid=f7fba2a96004431c6e3b90fb0728bd89")
+            let dataTask = URLSession.shared.dataTask(with: url!) { (data, response, error)  in
                 guard let data = data, error == nil else {
                     print("Error : \(error!.localizedDescription)")
+                    completionHandler(nil)
                     return
                 }
                 var fetchedWeatherData: WeatherInfoData?
```

Comment:
Is there any reason to declare this variable here? you can consider declare it as let inside do scope, as you're not changing it later on